local-cli/server/findSymlinksPaths.js

```diff
@@ -4,11 +4,34 @@ const fs = require('fs');
 module.exports = function findSymlinksPaths(lookupFolder) {
   const timeStart = Date.now();
   const folders = fs.readdirSync(lookupFolder);
-  const resolvedSymlinks = folders.map(folder => path.resolve(lookupFolder, folder))
-    .filter(folderPath => fs.lstatSync(folderPath).isSymbolicLink())
-    .map(symlink => path.resolve(process.cwd(), fs.readlinkSync(symlink)));
-  const timeEnd = Date.now();
 
+  const resolvedSymlinks = [];
+  folders.forEach(folder => {
+    const visited = [];
+
+    let symlink = path.resolve(lookupFolder, folder);
+    while (fs.lstatSync(symlink).isSymbolicLink()) {
+      const index = visited.indexOf(symlink);
+      if (index !== -1) {
+        throw Error(
+          `Infinite symlink recursion detected:\n  ` +
+            visited.slice(index).join(`\n  `)
+        );
+      }
+
+      visited.push(symlink);
+      symlink = path.resolve(
+        path.dirname(symlink),
+        fs.readlinkSync(symlink)
+      );
+    }
+
+    if (visited.length) {
```

Comment:
Since we no longer call `folders.filter` to guarantee each item is a symlink, we need to protect against `while (fs.lstatSync(symlink).isSymbolicLink())` doing zero loops.